Table of contents

rupertj opened the pull request:
Adds table of contents block, along with other things like a composer.json file and CI setup.

finnlewis commented:
Hey @rupertj ! 

Thanks for this, just taking it for a spin.

One initial point is that we have some (recently updated) branch naming conventions. This is certainly not a criticism, just sharing for general awareness.

We like to name feature branches something like feature/4-table-of-contents

See 
https://github.com/localgovdrupal/localgov/wiki/Git-branch-naming-conventions

Let me know if you have any questions or comments about that, keen to know how it reads with fresh eyes!

Cheers,

Finn

finnlewis commented on config/install/core.entity_form_display.node.publication.default.yml:
```diff
@@ -0,0 +1,84 @@
+    region: content
+    settings: {  }
+    third_party_settings: {  }
+  field_localgov_publication_conte:
```
It would be good to name this field without the field_ prefix. 

For changing the default in Drupal, see https://github.com/localgovdrupal/localgov/wiki/Naming-conventions#set-field-prefix-to-localgov_

finnlewis commented on config/install/field.field.node.publication.field_localgov_publication_conte.yml:
```diff
@@ -0,0 +1,55 @@
+  handler: 'default:paragraph'
+  handler_settings:
+    target_bundles:
+      from_library: from_library
```
The from_library paragraph type is used on service pages mainly for re-usable contact blocks, where they want to have one place to update contact details and it be updated across all instances. 

In general, I think we might want to avoid encouraging paragraph re-use unless specifically needed.

In the case of publications, I think we probably don't want to include it.

finnlewis replied:
Any thoughts on this one? Is it a real user need to have reusable paragraphs in a publication?

finnlewis commented on src/Plugin/Block/TocBlock.php:
```diff
@@ -0,0 +1,126 @@
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
```
On a fresh install, I had to place this block, it would be good to have it placed for common themes (localgov_base / localgov_scarfolk?)

We've done this with config/optional on other modules like step-by-step, for example see: 
https://github.com/localgovdrupal/localgov_step_by_step/blob/2.x/config/optional/block.block.localgov_step_by_step_navigation_base.yml

finnlewis commented:
Thanks @rupertj !

I tested this on a fresh install I had to place the 'table of contents' block manually, but once done, it did work as advertised. 

A few suggestions above, but once addressed we should get this merged to the 1.x branch so we can have the composer.json and add this to packagist for easier composer based install.

rupertj commented:
Thanks for the review and the pointer to the branch name docs @finnlewis. Do you want me to push up a correctly-named branch? I'm thinking as this one's (hopefully) ready to merge it might be easier to just do that - but I'm happy to make a new one if you like.

That doc reads just fine btw.

I've added some more config for the block placement to this branch, and renamed the field I added to the publication type to follow the other LGD conventions.

finnlewis reviewed (approved):
This looks good to me for now, still not sure of the need for reusable paragraphs, so we might want to check on that further down the line.

Thanks @rupertj
